classes/article/ArticleGalleyDAO.inc.php

```diff
@@ -235,27 +248,49 @@ function _fromRow($row) {
 	function insertObject($galley) {
 		$this->update(
 			'INSERT INTO submission_galleys
-				(submission_id, label, locale, seq, remote_url, file_id)
+				(submission_id, label, locale, seq, remote_url)
 				VALUES
-				(?, ?, ?, ?, ?, ?)',
+				(?, ?, ?, ?, ?)',
 			array(
 				(int) $galley->getSubmissionId(),
 				$galley->getLabel(),
 				$galley->getLocale(),
 				$galley->getSequence() == null ? $this->getNextGalleySequence($galley->getSubmissionId()) : $galley->getSequence(),
 				$galley->getRemoteURL(),
-				$galley->getFileId(),
 			)
 		);
 		$galley->setId($this->getInsertId());
 		$this->updateLocaleFields($galley);
+		$this->addFile($galley->getId(), $galley->getFileId(), $galley->getCurrentVersionId());
```

Comment:
Because you can only add galleys to the most recent version it is always the recent version id (Old versions can not be edited). Maybe this should be made clearer a the comments of the function ...